File: Algorithm/problem/baekjoon1816.py
T = int(input())

for testcase in range(1, T+1):
    number = int(input())

    for i in range(2, 1000001):
        if number % i == 0:
            print("NO")
            break

        if number == 1000000:
            print("YES")
            
    

File: Algorithm/problem/23_수학/Baekjoon1816.java
package Algorithm;

import java.io.BufferedReader;
import java.io.InputStreamReader;

public class Baekjoon1816 {
    public static void main(String[] args) throws Exception {

      BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

      int input = Integer.parseInt(br.readLine());

      for(int i=0; i < input; i++){
          long num = Long.parseLong(br.readLine());
          boolean bool = true;

            for(int j=2; j<1000001; j++){
              if(num % j == 0){
                bool = false;
              }
              if(i==1000000){
                bool = true;
              }
            }
              if(bool==false){
                System.out.println("NO");
              }
              else{
                System.out.println("YES");
              }
          }
  }
}
